analyzer/analyzer.spicy

```diff
@@ -467,36 +467,52 @@ type ImportLookupTable = unit(f: ImageFile) {
 };
 
 type ImportLookupTableEntry = unit(f: ImageFile) {
+    var full: uint64;
+    var importByName: bool;
+    var ordinal: uint16;
+    var hintNameRVA: uint32;
+
     switch ( f.peFileFormat ) {
-        PE32      -> entry: bitfield(32) {
-            full:         0..31 &convert=cast<uint64>($$);
-            importByName: 31    &convert=cast<bool>(! $$);
-            ordinal:      0..15 &convert=cast<uint16>($$);
-            hintNameRVA:  0..30 &convert=cast<uint32>($$);
-        };
-        PE32_Plus -> entry: bitfield(64) {
-            full:         0..63 &convert=cast<uint64>($$);
-            importByName: 63    &convert=cast<bool>(! $$);
-            ordinal:      0..15 &convert=cast<uint16>($$);
-            hintNameRVA:  0..30 &convert=cast<uint32>($$);
-        };
+        PE32      -> : bitfield(32) {
+            full:         0..31;
+            importByName: 31;
+            ordinal:      0..15;
+            hintNameRVA:  0..30;
+        } {
+            self.full = cast<uint64>($$.full);
+            self.importByName = cast<bool>(! $$.importByName);
+            self.ordinal = cast<uint16>($$.ordinal);
+            self.hintNameRVA = cast<uint32>($$.hintNameRVA);
+        }
+
+        PE32_Plus -> : bitfield(64) {
+            full:         0..63;
+            importByName: 63;
+            ordinal:      0..15;
+            hintNameRVA:  0..30;
+        } {
+            self.full = cast<uint64>($$.full);
+            self.importByName = cast<bool>(! $$.importByName);
+            self.ordinal = cast<uint16>($$.ordinal);
+            self.hintNameRVA = cast<uint32>($$.hintNameRVA);
+        }
     };
 
+    : void
+        {
+        if ( self.importByName )
+            self.hintNameOffset = rvaToOffset(f, self.hintNameRVA);
+        }
+
```

Comment:
FYI @rsmmr, we now(I think correctly) reject this since one of the bitfield cleanup commits. Not sure it would be generally useful, but what I would have wished for here was a way to map these indentical bitfield entries somehow into the parent unit so I wouldn't need to do that manually via the unit vars and the copy-n-paste code. Implementing this would likely require a runtime switch to pick values from the correct set bitfield though, so it might even be too expensive.

Reply:
Actually this is a regression: it's been intentional to support multiple switch cases with the same name as long as their types match. Sounds like that broke for bitfields now. Plus, I agree: I think natural thing to do would be mapping such anonymous bitfield cases into the parent scope. (I don't see what would need to be picked at runtime, am I missing something?).

Reply:
The types of the cases differ (`bitfield(32)` vs `bitfield(64)`). I would have thought that for different types we would need to figure out at runtime which field was set, but I am not 100% certain for what exact code we generate.

Reply:
Oh, I need more coffee, I missed the different widths. Yes, it's fine to reject, and I agree that we can't easily level them up.

Reply:
It looks like even if the types match, mapping across a unit `switch` is broken, zeek/spicy#1503.